lib/TPP/RewriteToBatchReduceGemm.cpp

```diff
@@ -92,18 +92,13 @@ static LogicalResult checkVNNIAccessPatterns(linalg::LinalgOp linalgOp) {
     if (map.getNumResults() < 3)
       return failure();
     if (operand->getOperandNumber() == 1) {
-      if (operand->get().getType().cast<ShapedType>().getRank() == 5 &&
-          map.getNumResults() != 5)
-        return failure();
       maps.push_back(map.getMinorSubMap(4));
     } else {
       maps.push_back(map.getMinorSubMap(3));
     }
   }
   for (OpOperand *operand : linalgOp.getDpsInitOperands()) {
     AffineMap map = linalgOp.getMatchingIndexingMap(operand);
-    if (map.getNumResults() < 2)
-      return failure();
     maps.push_back(map.getMinorSubMap(2));
```

Comment:
`maps.push_back(linalgOp.getMatchingIndexingMap(operand).getMinorSubMap(2));`

I think we can drop the temporary variable `map`.